src/bandersnatch/tests/test_mirror.py

```diff
@@ -1297,5 +1300,170 @@ def test_write_simple_pages(mirror: BandersnatchMirror) -> None:
+
+    # sanity check - we used monkeypatch to change cwd,
+    assert Path.cwd() == tmp_path
+
+
+# This checks the path where the diff-file is created when:
+# 1) `diff-file` points to an existing directory
+# 2) `diff-append-epoch` is enabled
+# One might expect any of:
+# a) `.../{diff-file}/mirrored-files-{timestamp}`
+# b) `.../{diff-file}-{timestamp}/mirrored-files`
+# c) `.../{diff-file}-{timestamp}`
+# The existing behavior is (c). Though this ignores the existing directory, changes to
+# diff-file path handling behavior could break a user's existing automation/scripting.
+@pytest.mark.asyncio
+async def test_mirror_subcommand_diff_file_dir_with_epoch(
+    tmp_path: Path, monkeypatch: pytest.MonkeyPatch
+) -> None:
+
+    mirror_dir = tmp_path / "mirror"
+    diff_files_dir = mirror_dir / "diff-files-dir"
+
+    # Setup: make sure the 'diff-file' directory is already created
+    diff_files_dir.mkdir(parents=True)
+
+    # Setup: create configuration for 'mirror' subcommand that includes both
+    # `diff-file` and `diff-append-epoch`
+    config = ConfigParser()
+    config.read_dict(
+        {
+            "mirror": {
+                "master": "https://pypi.org",
+                "directory": mirror_dir.as_posix(),
+                "timeout": 15,
+                "stop-on-error": False,
+                "workers": 1,
+                "hash-index": False,
+                "release-files": True,
+                "json": False,
+                "diff-file": diff_files_dir.as_posix(),
+                "diff-append-epoch": True,
+            },
+            "plugins": {"enabled": ["allowlist_project"]},
+            "allowlist": {"packages": ["black", "ruff", "autopep8"]},
+        }
+    )
+
+    # Setup: mock the 'synchronize' method on BandersnatchMirror to return fixed output.
+    # The output of the synchronize method is a mapping from package names to changed
+    # files, so the paths returned here should end up in diff-file.
+    mock_synchronize = make_mock_synchronize(
+        {
+            "black": {"AB/C1/ABC123/black-123-py3-non-any.whl"},
+            "ruff": {
+                "98/7D/987DEF/ruff-123.tar.gz",
+                "98/7D/987DEF/ruff-123-py3-none-any.whl",
+            },
+        }
+    )
+    monkeypatch.setattr(BandersnatchMirror, "synchronize", mock_synchronize)
+
+    # Setup: mock time.time so we can assert against the generated diff-file path
+    fixed_time = time.time()
+    monkeypatch.setattr(time, "time", lambda: fixed_time)
+
+    # Execute: run the 'mirror' subcommand
+    await mirror_cmd(config=config)
+
+    # Expectation: current behavior appends timestamp before checking if the directory exists,
+    # so the generated path will be `{diff-file}-{timestamp}`.
+    expected_diff_file = diff_files_dir.with_name(
+        f"{diff_files_dir.name}-{int(fixed_time)}"
+    )
+    assert expected_diff_file.is_file()
+    lines = expected_diff_file.read_text().splitlines()
+    assert len(lines) > 0
```

Comment:
```suggestion
    assert len(lines) == 3
```
This should always be the same number of lines right? Or did I count wrong / miss why > 0 is better here?

Reply:
You didn’t miss anything, I think this was just laziness 😆 And after already mocking `synchronize` too.

I’ll double-check this and update it.

Reply:
Good old Windows, I wonder if anything goes wrong on Linux and if you fix doesn't impact Linux/MacOS (POSIX filesystems) then I am cool for the fix to come in this PR.

Reply:
I can do a before/after check in a docker container this weekend to make certain, since if it did affect Linux I'm not sure there's an existing unit test that would catch the regression in CI.